addon-test-support/setup-middleware-reporter.ts

```diff
@@ -76,7 +76,8 @@ export async function middlewareReporter(axeResults: AxeResults) {
 
   if (!currentTestResult) {
     let { module, testName } = QUnit.config.current;
-    let testMetaData = getTestMetadata(getContext());
+    const context = getContext() ?? {};
+    let testMetaData = getTestMetadata(context);
```

Comment:
This looks reasonable from a types POV, but that's because while the types in `@ember/test-helpers` in this spot are not *wrong*, they are also not actively preventing us from ending up in a bad spot. If there is no `testContext` here, I expect this may work at a surface level, but that means nothing has correctly called one of `setupTest()`, `setupRenderingTest()`, or `setupApplicationTest()`, all of which call `setupContext()`. Which is… a weird spot to be in!

In the previous version of this, if that ever happened, we would get an error when running the tests, because under the hood, [it's going to try to set up a key-value pair on a `WeakMap`][getTestMetadata]:

[getTestMetadata]: https://github.com/emberjs/ember-test-helpers/blob/c25c5995ec60bc934740aaaf5d85299ede5ba0ee/addon/addon-test-support/@ember/test-helpers/test-metadata.ts#L41-L56

```ts
const TEST_METADATA = new WeakMap<BaseContext, TestMetadata>();

export default function getTestMetadata(context: BaseContext): TestMetadata {
  if (!TEST_METADATA.has(context)) {
    TEST_METADATA.set(context, new TestMetadata());
  }

  return TEST_METADATA.get(context)!;
}
```

If you do that and `getContext()` returns `undefined`, then you will see something like:

> TypeError: WeakMap keys must be objects or non-registered symbols

Net, what we should do here is throw a useful error that tells the end user that they need to make sure to have called `setupTest` etc. first:

```suggestion
    const context = getContext();
    if (!context) {
      throw new Error(
        'You tried to run ember-a11y-testing without calling one of the `setupTest` helpers from `@ember/test-helpers`. Please make sure yoru test setup calls `setupTest()`, `setupRenderingTest()`, or `setupApplicationTest()`!'
      );
    }
    let testMetaData = getTestMetadata(context);
```

Reply:
Sounds reasonable.